pkg/agent/baker_test.go

```diff
@@ -350,6 +350,14 @@ var _ = Describe("Assert generated customData and cseCmd", func() {
 		)
 		Expect(err).To(BeNil())
 		customDataBytes, err := base64.StdEncoding.DecodeString(nodeBootstrapping.CustomData)
+		Expect(err).To(BeNil())
+
+		// optionally unzip the bytes
```

Comment:
does it need to be optional?

Reply:
Interesting question. I made it optional so that the test runner would better match what the cloud-init does - where it is optional. However, we want all custom data to always be zipped from now on, so making it mandatory in the test runner gives us unit testing that they are zipped. So I've changed this to make the zipping mandatory.